src/layers/models/fileValidator.ts

```diff
@@ -113,4 +141,52 @@ export class FileValidator {
     });
     return allValid;
   }
+
+  private validateGpkgIndex(files: string[], originDirectory: string): void {
+    files.forEach((file) => {
+      const sqliteClient = new SQLiteClient(this.config, this.logger, file, originDirectory);
+      const index = sqliteClient.getGpkgIndex();
+      if (!index) {
+        const message = `Geopackage name: ${file} does not have a tiles index`;
+        this.logger.error({
+          originDirectory: originDirectory,
+          fileName: file,
+          msg: message,
+        });
+        throw new BadRequestError(message);
+      }
+    });
+  }
+
+  private validateGpkgGrid(files: string[], originDirectory: string): void {
+    files.forEach((file) => {
+      const sqliteClient = new SQLiteClient(this.config, this.logger, file, originDirectory);
+      const grid: Grid | undefined = sqliteClient.getGrid();
+      if (grid !== Grid.TWO_ON_ONE) {
+        const message = `Geopackage name: ${file} grid is not two_on_one`;
+        this.logger.error({
+          originDirectory: originDirectory,
+          fileName: file,
+          msg: message,
+        });
+        throw new BadRequestError(message);
+      }
+    });
+  }
+
+  private validateTilesWidthAndHeight(files: string[], originDirectory: string): void {
+    files.forEach((file) => {
```

Comment:
add try catch to this method and in the catch throw message "failed to validate Tiles size"